// @capacitor/electron/dist/esm/electron/splashscreen.js
import * as tslib_1 from "tslib";
import { WebPlugin } from "@capacitor/core";
var SplashScreenPluginElectron = /** @class */ (function (_super) {
    tslib_1.__extends(SplashScreenPluginElectron, _super);
    function SplashScreenPluginElectron() {
        var _this = _super.call(this, {
            name: 'SplashScreen',
            platforms: ['electron']
        }) || this;
        _this.ipc = null;
        _this.ipc = require('electron').ipcRenderer;
        return _this;
    }
    SplashScreenPluginElectron.prototype.show = function (options, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                this.ipc.send('showCapacitorSplashScreen', tslib_1.__assign({}, options));
                if (callback) {
                    callback();
                }
                return [2 /*return*/];
            });
        });
    };
    SplashScreenPluginElectron.prototype.hide = function (options, callback) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            return tslib_1.__generator(this, function (_a) {
                this.ipc.send('hideCapacitorSplashScreen', tslib_1.__assign({}, options));
                if (callback) {
                    callback();
                }
                return [2 /*return*/];
            });
        });
    };
    return SplashScreenPluginElectron;
}(WebPlugin));
export { SplashScreenPluginElectron };
var SplashScreen = new SplashScreenPluginElectron();
export { SplashScreen };
